### ISSUE TITLE
Improve interoperablilty of MDTraj/MDAnalysis trajectory handling

### ISSUE THREAD

**Issue by xiki-tempula** [michellab/BioSimSpace#340]
Say if I create a BSS.Trajectory.Trajectory object with

```
traj = Trajectory(None,
    "test/Sandpit/Exscientia/input/protein_ligand/traj.xtc",
    "test/Sandpit/Exscientia/input/protein_ligand/complex.tpr")
```
Then I cannot call traj.getFrames() as traj._trajectory is None

**Comment by xiki-tempula** [comment]
Ok, I see the issue, by default, BSS.Trajectory.Trajectory will use mdtraj to read the topology and thus, it will fail to read the Gromacs tpr file as topology, which is only supported by MDAnalysis.

**Comment by lohedges** [comment]
Yes, that's right. I'll need to add a check for a `tpr` topology extension and use MDAnalysis as the backend in that situation. (This is handled automatically when using a `Process`, since it knows what the engine is.)

**Comment by lohedges** [comment]
We've also had situations where some of our users don't want MDAnalysis in their environment since it conflicts with other packages, hence defaulting to MDTraj.

**Comment by xiki-tempula** [comment]
@lohedges Thanks, the whole getFrames() method might also need dome polishing if one assumes that the traj._trajectory could be an MDA object.

**Comment by lohedges** [comment]
I'll add some functionality that tries to load the files using either backend, then flag internally which one worked. Then we can simply use the same engine when extracting frames, etc.

**Comment by lohedges** [comment]
Closing for now as this has been addressed in #353. Feel free to open another thread if other trajectory handling issues arise.

Cheers.
